espresso/environment/2_espresso_liveness_test.go

```diff
@@ -0,0 +1,120 @@
+// the Dev node API at the moment.  However, we do have the ability to turn
+// the consensus layer on and off via turning hotshot on and off.
+//
+// This is **NOT** the same thing, nor would it result in the same behavior as
+// turning the Builder off and on. For the following reasons:
+//
+//	1 HotShot being off means no new blocks are being produced
+//	2 The Builder being off means that only empty blocks are being produced
+//	3 Turning the Builder off potentially means losing pool information,
+//	  requiring re-submission so that the builder can include the transaction
+//	  in the next block.
+//
+// With these caveats in mind, we may be able to simulate the behavior of 2
+// at the very least, if we intercept the client submitting transactions to
+// Espresso, and simulating the client being unable to submit transactions.
+// Likewise, we might be able to simulate 3 by falsely reporting to the
+// submitter that the transaction was submitted successfully, and withholding
+// the submission itself.
+func TestE2eDevNetWithEspressoEspressoDegradedLiveness(t *testing.T) {
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+
+	launcher := new(env.EspressoDevNodeLauncherDocker)
+
+	// Start a Server to proxy requests to Espresso
+	_, server, option := env.SetupQueryServiceIntercept(
+		// This decider will randomly report successful submissions of
+		// transactions to Espresso, but will not actually submit them.
+		// This will approximately occur 10% of the time, given the
+		// criteria to roll a number 0-9 and only to occur if the rolled
+		// number is 0.
+		env.SetDecider(env.NewRandomRollFakeSubmitTransactionSuccess(
+			10,
+			0,
+			rand.New(rand.NewSource(0)),
+		)),
+	)
+
+	defer server.Close()
+	system, espressoDevNode, err := launcher.StartDevNet(ctx, t, 0, option)
+
+	// Signal the testnet to shut down
+	if have, want := err, error(nil); have != want {
+		t.Fatalf("failed to start dev environment with espresso dev node:\nhave:\n\t\"%v\"\nwant:\n\t\"%v\"\n", have, want)
+	}
+
+	defer system.Close()
+	defer espressoDevNode.Stop()
+
+	addressAlice := system.Cfg.Secrets.Addresses().Alice
+
+	l2Seq := system.NodeClient(e2esys.RoleSeq)
+	l2Verif := system.NodeClient(e2esys.RoleVerif)
+
+	balanceAliceInitial, err := l2Verif.BalanceAt(ctx, addressAlice, nil)
+	if have, want := err, error(nil); have != want {
+		t.Fatalf("Failed to fetch Alice's balance:\nhave:\n\t\"%v\"\nwant:\n\t\"%v\"\n", have, want)
+	}
+
+	const N = 10
+	{
+		var receipts []*geth_types.Receipt
+
+		for i := 0; i < N; i++ {
+			receipt := helpers.SendL2TxWithID(t, system.Cfg.L2ChainIDBig(), l2Seq, system.Cfg.Secrets.Bob, func(opts *helpers.TxOpts) {
+				opts.Nonce = uint64(i)
+				opts.ToAddr = &addressAlice
+				opts.Value = new(big.Int).SetUint64(1)
+			})
+
+			receipts = append(receipts, receipt)
+		}
+
+		// Let's verify that all of our transactions came through successfully
```

Comment:
This seems redundant as [SendL2TxWithID](https://github.com/EspressoSystems/optimism-espresso-integration/blob/26cd5efe8acb2afd099dceb5c6e7c36cf5820da5/op-e2e/system/helpers/tx_helper.go#L91) already checks if the receipt is `types.ReceiptStatusSuccessful`.

Reply:
This is true, but it only verified them on the l2Seq, as not validator nodes were passed to SendL2TxWithID. This second section, is verifying that they exist on the l2Verif as well, which would indicate that the state is derivable without reading from the l2Seq.

Reply:
I see, so this is how we should send the transactions without waiting for the "long" confirmation. Thanks.